### PR TITLE
[AIRFLOW-3826][WIP] Group SageMaker sensors and operators into 2 files

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3826
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

This is the second attempt to clean up in Airflow.  The first attempt was made by @kaxil, but was delayed until the release of Airflow 2.0. @kaxil  solution was not backward compatible.
See: #3000 

My solution is backwards compatible. The user can use old and new paths at the same time.
In the future, support for old paths may be abandoned.

This is not the last version of the patch. For example: this fix requires `UPDATING.md` to be completed, but I would like to hear your feedback.

It is worth mentioning that IDE have support for notifications for the developer. The message will not be displayed only at runtime, but also when writing.

screenshot 2019-02-07 at 21 36 22

On this PR, please look and think about the whole platform, not as a single and simple PRs. In the future, further improvements can be made for other areas in a similar way. This solution also applies to the operators and sensors for the Google Cloud Platform.

Another solution is to create sub-modules.
The directory structure would then look that way.
````
+ airflow
+---+ contrib
|   +---+ sensors
|   |   +--- __init__.py
|   |   +--- gcp_transfer_operator.py
|   |   +--- gcp_sql_operator.py
|   |   +--- sagemaker_base_operator.py
|   |   +--- sagemaker_endpoint_config_operator.py
|   |   +--- sagemaker_endpoint_operator.py
|   |   +--- sagemaker_model_operator.py
|   |   +--- sagemaker_training_operator.py
|   |   +--- sagemaker_transform_operator.py
|   |   +--- sagemaker_tuning_operator.py
|   |   +--+ sagemaker_operator
|   |      +--- __init__.py
|   |      +--- base_operator.py
|   |      +--- endpoint_config_operator.py
|   |      +--- endpoint_operator.py
|   |      +--- model_operator.py
|   |      +--- training_operator.py
|   |      +--- transform_operator.py
|   |      +--- tuning_operator.py
|   +---- operators
````
With this structure, new and old imports  continued to work
```python
from airflow.contrib.operators.sagemaker_operator import SageMakerEndpointOperator
from airflow.contrib.operators.sagemaker_endpoint_operator import SageMakerEndpointOperator
```
After the transition period, we can delete the `sagemaker_*. py` files and then only new import path will work.

CC: @criccomini, @fenglu-g, @ashb 

### Tests

No applicable

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

Documentation updated

### Code Quality

- [X] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/sagemaker_base_operator.py [hunk 1]
```diff
@@ -17,84 +17,11 @@
...
-            'After preprocessing the config is:\n {}'.format(
-                json.dumps(self.config, sort_keys=True, indent=4, separators=(',', ': ')))
-        )
-
-    def execute(self, context):
-        raise NotImplementedError('Please implement execute() in sub class!')
+warnings.warn("The operator `SageMakerBaseOperator` has been moved to another "
+              "module. You should write "
+              "'from airflow.contrib.operators.segamaker_operator import SageMakerBaseOperator`"
+              " to load this operator.", DeprecationWarning)
```
Comment (ashb) [inline airflow/contrib/operators/sagemaker_base_operator.py:27]: What do you get when you run this with a file such as

```
form airflow.contrib.operators.sagemaker_base_operator import SageMakerBaseOperator
```

Specifically what line and file is this warning reported from? Normally with this kind of warning it's worth add `stacklevel=2` (I think) to report the warning from the callers perspective, but I'm not sure that works for imports.

The other thing to check out is `zope.deprecation` which is a module we already use in Airflow

File: airflow/contrib/sensors/sagemaker_sensor.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
+# -*- coding: utf-8 -*-
```
Comment (ashb) [inline airflow/contrib/sensors/sagemaker_sensor.py:1]: Thought about the file naming, specifically the `_sensor` suffix:

- Should it be `_sensors` as it contains more than one?
- Do we even need it at all given it's already under `airflow.contrib.sensors`?

Additionally: while we are moving classes around: lets move everything out of contrib, as there is no difference from contrib and not under Apache Airflow (but there was a difference under AirBnB Airflow)

**Comment by ashb** [comment]
I like it!

File: airflow/contrib/operators/sagemaker_base_operator.py (hunk 1 above)
Comment (ashb) [reply airflow/contrib/operators/sagemaker_base_operator.py:27]: https://zopedeprecation.readthedocs.io/en/latest/api.html#moving-modules

Comment (mik-laj) [reply airflow/contrib/operators/sagemaker_base_operator.py:27]: > The other thing to check out is `zope.deprecation` which is a module we already use in Airflow

`warnings` is module from Python. `zone.deprecation` uses `warnings` internally.

Comment (mik-laj) [reply airflow/contrib/operators/sagemaker_base_operator.py:27]: I will look deeper at the `stacklevel` option. Thank you for the suggestion.

File: airflow/contrib/operators/sagemaker_transform_operator.py [hunk 3]
```diff
@@ -17,108 +17,11 @@
...
-                'Model': self.hook.describe_model(
-                    transform_config['ModelName']
-                ),
-                'Transform': self.hook.describe_transform_job(
-                    transform_config['TransformJobName']
-                )
-            }
+warnings.warn("The operator `SageMakerTransformOperator` has been moved to another "
+              "module. You should write "
+              "'from airflow.contrib.operators.segamaker_operator import SageMakerTransformOperator`"
```
Comment (mik-laj) [inline airflow/contrib/operators/sagemaker_transform_operator.py:26]:
```suggestion
              "'from airflow.contrib.operators.sagamaker_operator import SageMakerTransformOperator`"
```

File: airflow/contrib/sensors/sagemaker_transform_sensor.py [hunk 4]
```diff
@@ -17,46 +17,11 @@
...
-
-        self.log.info('Poking Sagemaker Transform Job %s', self.job_name)
-        return sagemaker.describe_transform_job(self.job_name)
-
-    def get_failed_reason_from_response(self, response):
-        return response['FailureReason']
-
-    def state_from_response(self, response):
-        return response['TransformJobStatus']
+warnings.warn("The sensor `SageMakerTransformSensor` has been moved to another "
```
Comment (mik-laj) [inline airflow/contrib/sensors/sagemaker_transform_sensor.py:24]: Other proposition: `
> `SageMakerTransformSensor` has been moved to `airflow.contrib.sensors.sagemaker_sensor.SageMakerTransformSensor`. Your code should be updated to include `from airflow.contrib.sensors.sagemaker_sensor import SageMakerTransformSensor`.

**Comment by kaxil** [comment]
+1. Good job @mik-laj

File: airflow/contrib/operators/sagemaker_base_operator.py [hunk 5]
```diff
@@ -17,84 +17,11 @@
...
-        self.log.info(
-            'After preprocessing the config is:\n {}'.format(
-                json.dumps(self.config, sort_keys=True, indent=4, separators=(',', ': ')))
-        )
-
-    def execute(self, context):
-        raise NotImplementedError('Please implement execute() in sub class!')
+warnings.warn("The operator `SageMakerBaseOperator` has been moved to another "
+              "module. You should write "
+              "'from airflow.contrib.operators.segamaker_operator import SageMakerBaseOperator`"
```
Comment (ashb) [inline airflow/contrib/operators/sagemaker_base_operator.py:26]:
```suggestion
              "'from airflow.contrib.operators.sagemaker_operator import SageMakerBaseOperator`"
```

File: airflow/contrib/operators/sagemaker_transform_operator.py (hunk 3 above)
Comment (ashb) [reply airflow/contrib/operators/sagemaker_transform_operator.py:26]: sagE :D

**Comment by ashb** [comment]
Unless anyone else has any strong thoughts on the matter I say we move this to a new module called `airflow.sensors.sagemaker` so the usage becomes

```
from airflow.sensors.sagemaker import SageMakerFooSensor
```

**Comment by mik-laj** [comment]
@ashb SageMaker is a service from Amazon Web Services. I think that we should emphasize this.

**Comment by ashb** [comment]
So `from airflow.sensors.aws_sagemaker import SageMakerFooSensor` pattern then?

(Main point of my suggestion was 1: no contrib/, 2: multiple sensors in the same module)

**Comment by mik-laj** [comment]
The discussion about solving this problem and similar has been started in the thread:
https://lists.apache.org/thread.html/c880ef89f8cb4a0240c404f9372615b998c4a4eeca342651927d596c@%3Cdev.airflow.apache.org%3E

**Comment by mik-laj** [comment]
I'm closing this PR. I wanted to start discussions about the need for changes in import paths. I never wanted to finish this change.

**Pull request closed without merging**